pkgs/development/tools/apksigner/default.nix

```diff
@@ -1,15 +1,90 @@
-{ runCommand
+{ lib
+, stdenv
+, fetchgit
+, openjdk17_headless
+, gradle
+, perl
 , makeWrapper
-, jre
-, build-tools
 }:
-let
-  tools = builtins.head build-tools;
-in
-runCommand "apksigner" {
-  nativeBuildInputs = [ makeWrapper ];
-} ''
-  mkdir -p $out/bin
-  makeWrapper "${jre}/bin/java" "$out/bin/apksigner" \
-    --add-flags "-jar ${tools}/libexec/android-sdk/build-tools/${tools.version}/lib/apksigner.jar"
-''
+
+stdenv.mkDerivation rec {
+  pname = "apksigner";
+  version = "33.0.1";
+
+  src = fetchgit {
+    name = pname;
```

Comment:
Naming the source like binary is confusing. I would suggest to drop name here.

Reply:
IIRC `fetchgit` uses the repo name as `name` so it's more confusing. And `fetchgit` uses `name` as the directory name instead of source. Without this the source root is apksig and gradle uses this as the project name. So it will build a apksig.jar.

Reply:
So if we change it to $pname-src it will be build as apksig-src.jar? Hmmm. Probably should be commented.